mobile/src/screens/chat/chat.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { type ReactNode } from 'react';
+import LinearGradient from 'react-native-linear-gradient';
+
+import PlusIcon from '#assets/img/icons/plus.svg';
+import {
+  Card,
+  InputSearch,
+  Link,
+  ScrollView,
+  Text,
+  View,
+} from '#libs/components/components';
+import { AppColor, RootScreenName } from '#libs/enums/enums';
+import { useEffect, useNavigation, useState } from '#libs/hooks/hooks';
+
+import { Badge } from './components/components';
+import mockedChats from './libs/constants/data.json';
+import { styles } from './styles';
+
+type MockChatItem = {
+  id: string;
+  title: string;
+};
+
+const onPress = (): void => {
+  // This function is intentionally left empty for mocking purposes.
+};
+
+const Chat: React.FC = () => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [filteredChats, setFilteredChats] = useState<MockChatItem[]>([]);
+  const navigation = useNavigation();
+
+  useEffect(() => {
+    navigation.setOptions({
+      headerTitle: ({ children }: { children: ReactNode }) => (
+        <View style={styles.headerTitleWrapper}>
+          <Text style={styles.headerTitle}>{children}</Text>
+          <Badge />
+        </View>
+      ),
+    });
+  }, [navigation]);
+
+  useEffect(() => {
+    const filteredItems = mockedChats.filter((item) =>
+      item.title.toLowerCase().includes(searchQuery.toLowerCase()),
+    );
+    setFilteredChats(filteredItems);
+  }, [searchQuery]);
```

Comment:
Let's create a hook called useSearch and put this logic here. Also please use useAppForm for searchQuery

Reply:
- Is this new hook supposed to take only the chats array or it should work with other types? I see on the markups we have  Meditation and Breathing Screen, where we can use it also.
- If it should work with other types where it would be better to put types?

Reply:
I've added a hook, as I understood it, but could you please check it I am not sure that I did it correctly. and also without  useAppForm.

Reply:
Nice questions:
1. You are completely right, provide generic type
2. Generic can extend from { id, title } for example
3. We should handle all form changes via useAppForm, not useState

Reply:
Done